daemon/rename.go

```diff
@@ -31,6 +35,11 @@ func (daemon *Daemon) ContainerRename(oldName, newName string) error {
 
 	container.Lock()
 	defer container.Unlock()
+
+	if oldName == newName {
```

Comment:
Just discussed this with @vdemeester and we saw some other things that may need to be improved (not related to this PR), so he's going to do a follow up